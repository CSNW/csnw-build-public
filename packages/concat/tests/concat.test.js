const { join } = require('path');
const { Files } = require('@csnw-build/pipeline');

const concat = require('../')({
  'bundle.js': ['a.js', 'b.js', 'c.js']
});
const cwd = join(__dirname, 'fixtures');

test('pipeline', async () => {
  const process = await concat.pipeline({
    cwd,
    sourcemaps: true,
    cache: false
  });
  const files = await Files.from('*.js', { cwd });

  const result = await process(files);
  expect(result).toMatchSnapshot();
});
