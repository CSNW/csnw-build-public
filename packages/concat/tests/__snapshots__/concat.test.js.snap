// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipeline 1`] = `
Object {
  "bundle.js": Object {
    "baseDir": "C:/dev/csnw-build/packages/concat/tests/fixtures",
    "contents": "var name = 'a';

var name = 'b';

var name = 'c';
",
    "cwd": "C:/dev/csnw-build/packages/concat/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/concat/tests/fixtures/bundle.js",
    "sourceMap": Object {
      "file": "bundle.js",
      "mappings": "AAAA;AACA;ACDA;AACA;ACDA;AACA",
      "names": Array [],
      "sources": Array [
        "a.js",
        "b.js",
        "c.js",
      ],
      "sourcesContent": Array [
        "var name = 'a';
",
        "var name = 'b';
",
        "var name = 'c';
",
      ],
      "version": 3,
    },
  },
}
`;
